airflow/www/utils.py

```diff
@@ -476,12 +476,12 @@ def datetime_html(dttm: DateTime | None) -> str:
     return Markup('<nobr><time title="" datetime="{}">{}</time></nobr>').format(as_iso, as_iso_short)
 
 
-def json_f(attr_name):
+def json_f(attr_name, json_encoder: type[json.JSONEncoder] = json.JSONEncoder):
     """Returns a formatted string with HTML for given JSON serializable."""
 
     def json_(attr):
         f = attr.get(attr_name)
-        serialized = json.dumps(f)
+        serialized = json.dumps(f, cls=json_encoder)
```

Comment:
I think we should always use WebEncoder here, so the extra argument is not needed.

Right now `json_f` is actually only used once, but since `json_f` is only for the web UI, all usages in the future should use WebEncoder anyway.

Reply:
Personally speaking I've typically found it easier to keep parameterization open since it's more flexible and we should probably not be so sure about future usage of `json_f`, but I am open to suggestions. Maybe we can wait for a second opinion?

Reply:
@uranusjr I've updated the code so that `json_f` always uses `WebEncoder` and updated the relevant func calls + unit tests. Please review at your convenience. Thank you!